### PR TITLE
Add rpk registry code examples

### PR DESCRIPTION
**Pull request by kbatuigas**
Adds rpk registry examples to the [Use the Schema Registry API](https://docs.redpanda.com/beta/manage/schema-reg/schema-reg-api/) doc.

Preview: https://deploy-preview-106--redpanda-docs-preview.netlify.app/23.3/manage/schema-reg/schema-reg-api/

The new rpk registry reference docs have been merged from a separate branch.

Resolves https://github.com/redpanda-data/documentation-private/issues/2084

### CODE REVIEW

File: modules/manage/pages/schema-registry.adoc [hunk 1]
```diff
@@ -736,6 +798,15 @@ res = requests.delete(f'{base_uri}/subjects/sensor-value/versions/1', params=pay
 pretty(res)
 ```
 
+--
+rpk::
```
Comment (kbatuigas) [inline modules/manage/pages/schema-registry.adoc]: @r-vasquez The [docs](https://docs.redpanda.com/current/manage/schema-registry/#hard-delete-a-schema) currently mention that to perform a hard delete, the API endpoint has to be called twice, with the second call including the "permanent" parameter. Does that also apply to the rpk command or is running the command once sufficient for a hard delete?

**Comment by kbatuigas** [comment]
@r-vasquez Are there supposed to be 1:1 equivalents with the existing API methods? Not sure what the corresponding rpk registry commands are for:
  - `GET /schemas/types`: https://docs.redpanda.com/current/manage/schema-registry/#query-supported-schema-formats
  -  `GET /subjects/{subject}/versions`: https://docs.redpanda.com/current/manage/schema-registry/#retrieve-schema-versions-of-a-subject
  - `GET /subjects/{subject}/versions/{version}` https://docs.redpanda.com/current/manage/schema-registry/#retrieve-a-schema-of-a-subject

File: modules/manage/pages/schema-registry.adoc (hunk 1 above)
Comment (r-vasquez) [reply modules/manage/pages/schema-registry.adoc]: In `rpk` we have a `--permanent` flag that does the 2 deletes for you:

```
rpk registry schema-delete <subject> --schema-version latest --permanent
```

Same for `subject`

**Comment by r-vasquez** [comment]
@kbatuigas 
- `GET /schemas/types` : we don't have an rpk equivalent.
- `GET /subjects/{subject}/versions:`: we don't have an rpk equivalent yet.
- `GET /subjects/{subject}/versions/{version}` You can retrieve a schema by subject and version:

```
$ rpk registry schema get test-1 --schema-version latest
SUBJECT  VERSION  ID    TYPE
test-1   3        3     AVRO
```

But currently rpk does not print the schema.

Cc @twmb What do you think of adding the second point to rpk?  `GET /subjects/{subject}/versions:`

**Comment by twmb** [comment]
+1 from me

**Comment by kbatuigas** [comment]
Thank you @r-vasquez and @twmb ! Do you think those rpk equivalents might be added soon? We actually have another PR that is separately working on reorganizing our Schema Registry docs, so I'm trying to help scope out how the rpk registry addition might impact that.

**Comment by r-vasquez** [comment]
@kbatuigas 
GET /schemas/types : Probably won't be coming to rpk.
GET /subjects/{subject}/versions: Probably after the next release, but is not prioritized yet.

File: modules/manage/pages/schema-reg/schema-reg-api.adoc [hunk 2]
```diff
@@ -207,6 +164,14 @@ res = requests.post(
 pretty(res)
 ```
 
+--
+rpk::
++
+--
+
+```bash
+rpk registry schema create sensor-value ~/code/tmp/sensor_sample.avro
```
Comment (kbatuigas) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]: @r-vasquez Can rpk accept the json/yaml data like with the curl and Python examples, or will it only take a path to a file?

Comment (r-vasquez) [reply modules/manage/pages/schema-reg/schema-reg-api.adoc]: For now it only takes a path to a file

File: modules/manage/pages/schema-reg/schema-reg-api.adoc [hunk 3]
```diff
@@ -207,6 +164,14 @@ res = requests.post(
 pretty(res)
 ```
 
+--
+rpk::
```
Comment (r-vasquez) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]: Is it possible to put the rpk tab first than the curl and python?

File: modules/manage/pages/schema-reg/schema-reg-api.adoc [hunk 4]
```diff
@@ -247,6 +212,15 @@ res = requests.get(f'{base_uri}/schemas/ids/1').json()
...
++
+--
+
+```bash
+rpk registry schema get --id 1
```
Comment (r-vasquez) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]: In rpk there are 3 ways to do this:

```
* By version, returning a schema for a required subject and version
  rpk registry schema get [subject] --schema-version <version>
  
* By ID, returning all subjects using the schema, or filtering for one subject
  rpk registry schema get --id <id>
* By schema, checking if the schema has been created in the subject
  rpk registry schema get [subject] --schema <path/to/file>
```

Maybe it's worth adding a note or linking to reference docs about this.

**Review by r-vasquez — COMMENTED** [review]
There are a lot of new files to this PR that I think doesn't belong to the PR.

I was reading the preview but found out that: https://deploy-preview-106--redpanda-docs-preview.netlify.app/current/manage/schema-reg/schema-reg-api/?tab=tabs-9-curl#compatibility-uses-and-constraints didn't have the rpk tab while rpk support this with

```
rpk registry compatibility-level set
```

Sorry to leave the comment here, I can't find the line in the PR :sweat_smile:

File: modules/manage/pages/schema-reg/schema-reg-api.adoc (hunk 2 above)
Comment (r-vasquez) [reply modules/manage/pages/schema-reg/schema-reg-api.adoc]: The path to the Avro file is passed via a flag: `--schema`

```suggestion
rpk registry schema create sensor-value --schema ~/code/tmp/sensor_sample.avro
```

File: modules/manage/pages/schema-reg/schema-reg-api.adoc (hunk 3 above)
Comment (kbatuigas) [reply modules/manage/pages/schema-reg/schema-reg-api.adoc]: @r-vasquez definitely, I'll make that change

File: modules/manage/pages/schema-reg/schema-reg-api.adoc (hunk 4 above)
Comment (kbatuigas) [reply modules/manage/pages/schema-reg/schema-reg-api.adoc]: @r-vasquez You should be able to see the rpk tabs as long as you set the docs version in the deploy preview to 23.3 Beta -- let me know if it doesn't work! 

For this particular example I just kept it consistent with the the curl and Python examples, which only retrieve the schema by ID (no subjects or versions). I also linked out to the Intro to rpk and rpk registry docs in the [Prerequisites](https://deploy-preview-106--redpanda-docs-preview.netlify.app/23.3/manage/schema-reg/schema-reg-api/#prerequisites) section. Does that work?

**Review by r-vasquez — APPROVED** [review]
LGTM Thanks a lot!

File: modules/manage/pages/schema-reg/schema-reg-api.adoc [hunk 5]
```diff
@@ -590,70 +725,104 @@ To soft delete a schema, make a DELETE request with the subject and version ID (
...
-Doing a soft delete for an already deleted schema returns an error. For example:
+Doing a soft delete for an already deleted schema returns an error:
 
+[tabs]
+====
+rpk::
++
+--
+```
+Subject 'sensor-value' Version 1 was soft deleted.Set permanent=true to delete permanently
```
Comment (Feediver1) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]:
```suggestion
Subject 'sensor-value' Version 1 was soft deleted. Set permanent=true to delete permanently
```

File: modules/manage/pages/schema-reg/schema-reg-api.adoc [hunk 6]
```diff
@@ -695,42 +870,68 @@ curl -s \
...
 pretty(res)
 payload = { 'permanent' : 'true' }
 res = requests.delete(f'{base_uri}/subjects/sensor-value/versions/1', params=payload).json()
 pretty(res)
 ```
-
 --
 ====
 
 If the requests are successful, each request returns the version ID of the deleted schema:
```
Comment (Feediver1) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]:
```suggestion
If the requests are successful, each request returns the version ID of the deleted schema:
```

I think this should omit the "If the requests are successful..."

Each time I read this line throughout the PR, I thought, what if it is _not_ successful? It suggests to me that there is some likelihood of it not being successful. Your call, but I'd remove that preface from each of the statements in which it appears throughout.

File: modules/manage/pages/schema-reg/schema-reg-api.adoc [hunk 7]
```diff
@@ -590,70 +725,104 @@ To soft delete a schema, make a DELETE request with the subject and version ID (
...
-
 ```python
 res = requests.delete(f'{base_uri}/subjects/sensor-value/versions/1').json()
 pretty(res)
 ```
-
 --
 ====
 
 If the request is successful, it returns the ID of the soft deleted schema:
```
Comment (Feediver1) [inline modules/manage/pages/schema-reg/schema-reg-api.adoc]:
```suggestion
Returns the ID of the soft deleted schema:
```

**Review by Feediver1 — APPROVED** [review]
See comments

**Pull request merged**